diff -Nurp TelegramQML.orig/telegramqml.cpp TelegramQML/telegramqml.cpp
--- TelegramQML.orig/telegramqml.cpp	2017-04-16 17:12:24.811718425 +0200
+++ TelegramQML/telegramqml.cpp	2017-04-16 17:14:28.523849528 +0200
@@ -5372,7 +5372,12 @@ void TelegramQml::refreshSecretChats()
     const QList<SecretChat*> &secrets = p->tsettings->secretChats();
     Q_FOREACH(SecretChat *sc, secrets)
     {
+
         EncryptedChat chat(EncryptedChat::typeEncryptedChat);
+
+        if (sc->date() == 0)
+            chat.setClassType(EncryptedChat::typeEncryptedChatWaiting);
+
         chat.setAccessHash(sc->accessHash());
         chat.setAdminId(sc->adminId());
         chat.setDate(sc->date());
