diff -Nurp TelegramQML.orig/telegramqml.cpp TelegramQML/telegramqml.cpp
--- TelegramQML.orig/telegramqml.cpp	2017-04-16 17:44:44.437480000 +0200
+++ TelegramQML/telegramqml.cpp	2017-04-16 17:46:09.341258602 +0200
@@ -4305,6 +4305,7 @@ void TelegramQml::uploadCancelFile_slt(q
 
 void TelegramQml::fatalError_slt()
 {
+    Q_EMIT fatalError();
     try_init();
 }
 
diff -Nurp TelegramQML.orig/telegramqml.h TelegramQML/telegramqml.h
--- TelegramQML.orig/telegramqml.h	2017-04-16 17:44:44.437480000 +0200
+++ TelegramQML/telegramqml.h	2017-04-16 17:46:42.837172949 +0200
@@ -471,6 +471,8 @@ Q_SIGNALS:
 
     void errorSignal(qint64 id, qint32 errorCode, QString errorText, QString functionName);
 
+    void fatalError();
+
 protected:
     void try_init();
 
